ichnaea/api/locate/cell.py

```diff
@@ -42,7 +43,7 @@
         ("age", numpy.int32),
         ("signalStrength", numpy.int32),
         ("score", numpy.double),
-        ("id", "S11"),
+        ("id_b64", "S16"),
```

Comment:
I've never used numpy. This changes the field from a 0-terminated bytes array of 11 bytes to a 0-terminated bytes array of 16 bytes, right? If the mac address is in hex and we're not really doing math on it (it's just a label), why not switch it to a U? Is it just space concerns? Are there other niceties of being a 0-terminated byte array?

Reply:
The short answer is that cell IDs are more complex than mac addresses, and a binary field keeps the matching algorithm generic enough to handle all the station types.

Yes, `S16` is a bytestring of length 16. I came up with 16 by base64-encoding an 11-character string. The numpy type appears to be a null-terminated C-string, with all the issues that brings up. See the numpy issue for more weirdness.

The binary string is used to match the unique database key for the row. For WiFi and bluetooth, it is the binary encoding of the MAC ID. For cell stations and areas, it is the binary packed version of the type, the mobile country code (mcc), the mobile network code (mnc), the location area code (lac), and a cell identifier (cid):

https://github.com/mozilla/ichnaea/blob/14b73b16c702abb1a3f8262330631e56e521a938/ichnaea/models/cell.py#L222-L237

By using the base64 encoded binary ID, we avoid dropping training null characters, and still compare using database IDs in ``InternalPositionSource``.